pkg/types/urlsmap.go

```diff
@@ -71,23 +74,19 @@ func (c URLsMap) Len() int {
 }
 
 // parse parses the given string and returns a map listing the values specified for each key.
-func parse(s string) map[string][]string {
+func parse(s string) (map[string][]string, error) {
 	m := make(map[string][]string)
-	for s != "" {
-		key := s
-		if i := strings.IndexAny(key, ","); i >= 0 {
-			key, s = key[:i], key[i+1:]
-		} else {
-			s = ""
-		}
-		if key == "" {
-			continue
-		}
-		value := ""
-		if i := strings.Index(key, "="); i >= 0 {
-			key, value = key[:i], key[i+1:]
+	if s == "" {
+		return m, nil
+	}
+	pairs := strings.Split(s, ",")
+	for _, pair := range pairs {
```

Comment:
> is there a reason we do not want to keep the previous logic

No special reason, I was just thinking using `strings.Split()` was more readable.  

BTW, is there a reason about the implementation of https://github.com/coreos/etcd/blob/master/pkg/fileutil/fileutil.go#L39

Implemented as
```
func IsDirWriteable(dir string) error {
	return syscall.Access(dir, syscall.O_RDWR)
}
```
is much more elegant, right?